Assets/Mirror/Runtime/NetworkIdentity.cs

```diff
@@ -75,13 +75,15 @@ public sealed class NetworkIdentity : MonoBehaviour
         /// <para>This value is determined at runtime. For most objects, authority is held by the server.</para>
         /// <para>For objects that had their authority set by AssignClientAuthority on the server, this will be true on the client that owns the object. NOT on other clients.</para>
         /// </summary>
+        bool isOwner;
+
         public bool hasAuthority { get; internal set; }
 
         /// <summary>
         /// The set of network connections (players) that can see this object.
         /// <para>null until OnStartServer was called. this is necessary for SendTo* to work properly in server-only mode.</para>
         /// </summary>
-        public Dictionary<int, NetworkConnection> observers;
+        public HashSet<NetworkConnection> observers;
```

Comment:
one thing to keep in mind is that some games might want to find a certain observer by connectionId each frame.
with hashset, this would become really expensive.

Reply:
I would love to get rid of connectionId completely actually.   I don't know why games would need it at all.

Reply:
Uhh, maybe because one want to have some sort of other logic that keeps track of connection specific data, like scoreboards, etc? That way the connection id can say "Okay, player 22 belongs to Team Red". How would you do that if you took away connectionId?

Reply:
you would put a "team"  field in the player component.